src/analysis/problematic_exclusions.py

```diff
@@ -0,0 +1,289 @@
+    df2[term_id_field] = df2[term_id_field].apply(uri_converter.compress)
+    term_label_map = {}
+    for _i, row in df2.iterrows():
+        term_label_map[row[term_id_field]] = row[term_label_field]
+
+    # JOIN results
+    df_joined = pd.DataFrame()
+    if len(df) > 0:
+        rows = []
+        for _i, row in df.iterrows():
+            d_row = dict(row)
+            d_row[term_label_field] = term_label_map.get(d_row[term_id_field], '')
+            rows.append(d_row)
+        df_joined = pd.DataFrame(rows)
+
+    return df_joined
+
+
+def run(
+    onto_path: str, onto_config_path: str, component_signature_path: str, mirror_signature_path: str,
+    mondo_mappings_path: str, outpath: str, save=CONFIG['save']
+) -> Dict[str, pd.DataFrame]:
+    """For a given ontology corresponding to the files passed, find excluded terms that still exist in Mondo xrefs"""
+    term_id_field = 'term_id'
+    term_label_field = 'term_label'
+    mirror_field = '1_in_mirror_tsv'
+    component_field = '2_in_component_tsv'
+    mappings_field = '3_in_mondo_xrefs'
+    analytical_field = 'in1_notIn2_in3'
+    summary_count_field = 'n_in1_notIn2_in3'
+    outpath_xref_summary = outpath.replace('.tsv', '_summary.tsv')
+    mondo_mappings_df = pd.read_csv(mondo_mappings_path, sep='\t', comment='#').fillna('')
+    mondo_mappings_set = set(mondo_mappings_df['object_id'])
+
+    # Prefixes
+    with open(onto_config_path, 'r') as stream:
+        onto_config = yaml.safe_load(stream)
+    prefix_uri_map = onto_config['base_prefix_map']
+
+    # Load datasets
+    mirror_df = load_and_format_tsv(path=mirror_signature_path, prefix_map=prefix_uri_map)
+    component_df = load_and_format_tsv(path=component_signature_path, prefix_map=prefix_uri_map)
+    # todo?: maybe using column '?term' is better since could be more columns w/ diff order, but I figured it was
+    #  ...more likely that the column name '?term' could change, so I simply picked out the assumed only column [0].
+    mirror_set = set(mirror_df[list(mirror_df.columns)[0]])
+    component_set = set(component_df[list(component_df.columns)[0]])
+
+    # Determine set membership
+    term_appearances = {}
+    for term in mirror_set:
+        term_appearances[term] = {
+            term_id_field: term, mirror_field: True, component_field: False, mappings_field: False}
+    for term in component_set:
+        if term not in term_appearances:
+            term_appearances[term] = {term_id_field: term, mirror_field: False, mappings_field: False}
+        term_appearances[term][component_field] = True
+    for term in mondo_mappings_set:
+        if term not in term_appearances:
+            term_appearances[term] = {term_id_field: term, mirror_field: False, component_field: False}
+        term_appearances[term][mappings_field] = True
+    df = pd.DataFrame(list(term_appearances.values()))
+
+    # Filter
+    df = df[df[mappings_field] == True]
```

Comment:
@matentzn Filtered so that rows where `'3_in_mondo_xrefs' == FALSE'` no longer appear in the tables. Updated tables and re-committed.

Reply:
Nico and I talked and we decided that this was indeed the case. Now filtering by `'n_in1_notIn2_in3' == True`.